src/geo/map-view.js

```diff
@@ -52,8 +55,14 @@ var MapView = View.extend({
   },
 
   addOverlay: function (overlay) {
+    var type;
     if (overlay) {
-      this.$el.append(overlay.render().el);
+      type = overlay.type;
+      if (type && CONTAINED_OVERLAYS.indexOf(type) >= 0) {
+        this._overlayContainer().append(overlay.render().el);
+      } else {
```

Comment:
What other types of overlays do we have? Why not add them all to the same container? Individual overlays can be positioned absolutely via CSS anyway...